fix(page-title-bar): fix header top margin and height in condensed mode

kevinsperrine opened the pull request:
Closes #3012 

**Summary**

- fixes the max height for a couple situations with the PageTitleBar

**Change List (commits, features, bugs, etc)**

- fix max height of base breadcrumbs
- fix max height of condensed mode
- fix max height of dynamic mode

**Acceptance Test (how to verify the PR)**

- check the [base story](https://deploy-preview-3033--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--base) and confirm the breadcrumbs are 32px tall (include padding)
- check the [with dynamic breadcrumbs actions and content story](https://deploy-preview-3033--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--with-dynamic-scrolling)
- scroll down to condense the header, confirm it's 40px tall and the buttons are even with the bottom
- check the [condensed with primary + secondary buttons story](https://deploy-preview-3033--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--with-condensed-header)
- scroll down to condense the header
- confirm the header is 40px tall and the buttons are even with the bottom

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check other PageTitleBar stories for regressions

<!-- For reviewers: do not remove -->

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

kevinsperrine commented on packages/react/src/components/OverflowMenu/_overflow-menu.scss:
```diff
@@ -3,3 +3,9 @@
+  &:after {
+    width: $spacing-09;
+  }
+}
```
accidentally based this PR on the branch with the overflow menu css fixes.

bjornalm commented:
Not sure if this is just a story bug but the "Take an action" button is not aligned above the bottom border in story https://deploy-preview-3033--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--with-dynamic-scrolling
![image](https://user-images.githubusercontent.com/5167091/140525543-ac9ed932-8045-4b6a-8ed1-d2f89b733f67.png)

kevinsperrine commented:
> Not sure if this is just a story bug but the "Take an action" button is not aligned above the bottom border in story https://deploy-preview-3033--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-pagetitlebar--with-dynamic-scrolling ![image](https://user-images.githubusercontent.com/5167091/140525543-ac9ed932-8045-4b6a-8ed1-d2f89b733f67.png)

Good eye. I never even see 1px differences. Fixed. It was a negative margin applied--which also was messing with the e2e test, so I adjusted those, too.

bjornalm reviewed (approved):
Visually it looks good, but I think the clamp is redundant.

kevinsperrine commented on packages/react/src/components/PageTitleBar/_page-title-bar.scss:
```diff
@@ -152,7 +152,7 @@
     align-items: center;
     padding: $spacing-04 0 $spacing-02 $spacing-07;
-    max-height: clamp(0rem, $spacing-02, $spacing-08);
+    max-height: clamp(0rem, #{$spacing-07}, #{$spacing-08});
```
Yeah, I don't know why it was ever added as clamp to begin with, but I assumed it had a reason and left it in place for some edge case rather than remove it. 🤷‍♂️

kevinsperrine replied:
removed the clamp. It seems to be fine. Can you give it another look after netlify deploys?

bjornalm replied:
Looks good
